src/Workspaces/CSharp/Portable/Extensions/SyntaxNodeExtensions_SharedWithCodeStyle.cs

```diff
@@ -1,24 +1,32 @@
 ﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
 
-using System;
+#nullable enable
+
 using System.Collections.Generic;
+using System.Diagnostics.CodeAnalysis;
 using System.Linq;
-using System.Text;
 using Microsoft.CodeAnalysis.CSharp.Formatting;
 using Microsoft.CodeAnalysis.CSharp.Syntax;
 using Microsoft.CodeAnalysis.Shared.Extensions;
 using Microsoft.CodeAnalysis.Text;
+using Roslyn.Utilities;
 
 namespace Microsoft.CodeAnalysis.CSharp.Extensions
 {
     internal static partial class SyntaxNodeExtensions
     {
-        public static bool IsKind<TNode>(this SyntaxNode node, SyntaxKind kind, out TNode result)
+        public static bool IsKind<TNode>([NotNullWhen(returnValue: true)] this SyntaxNode? node, SyntaxKind kind, [NotNullWhen(returnValue: true)] out TNode? result)
             where TNode : SyntaxNode
         {
             if (node.IsKind(kind))
             {
+#if !CODE_STYLE
```

Comment:
Personal preference is to avoid putting `!` and swap the if here. If we have this pattern elsewhere favor that (and let me know :) )

Reply:
This is being done later in the file as well. I agree it's ugly but I kept it consistent. @sharwell you're probably the one most likely to have an opinion -- mind if we switch them all around?

Reply:
Why not just use this?

```
result = (TNode?)node;
```

Reply:
I didn't terribly like that because it wasn't really a correct annotation for either branch, and by having the explicit #if made it clear that we're in a situation where annotations behave differently. I was about to file a bug against the compiler thinking everything was broken until I realized this was a side effect of us targeting an older version of ourselves.

I'm not sure if using your copy annotations tool for our reference of our own packages makes the most sense here -- as we further null annotate things we're only going to keep running into issues like this.

> It works because the compiler doesn't validate NotNullWhen conditions inside the implementation, and I'm not aware of any pending change to this.

Was going to file that feedback today as I really wish the compiler did do that. :smile:

Reply:
(I'm merging this PR so we don't get the bulk of the benefit blocked up on this particular issue. Happy to address further.)